cpp/src/arrow/ipc/message.h

```diff
@@ -144,33 +144,20 @@ class ARROW_EXPORT MessageReader {
  public:
   virtual ~MessageReader() = default;
 
+  /// \brief Create MessageReader that reads from InputStream
+  static std::unique_ptr<MessageReader> Open(io::InputStream* stream);
+
+  /// \brief Create MessageReader that reads from owned InputStream
+  static std::unique_ptr<MessageReader> Open(
```

Comment:
For the record, is there a rationale or convention for the use of unique_ptr vs shared_ptr here? :-)

Reply:
I try to only use shared_ptr when there is some reasonable expectation that shared ownership may be frequently needed (of course one can always transfer the pointer to a shared_ptr if needed). There are some other places in the library where `shared_ptr` is returned (or an out-variable) that would be better as `unique_ptr`